### PR TITLE
Deposits

### PR DESCRIPTION
**Pull request by DmytroOzzmos**
Done task for deposits

### CODE REVIEW

File: App.Deposits/DepositsManager.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+        public DepositsManager(IDepositsRepository depositsRepository)
+        {
+            this.depositsRepository = depositsRepository;
+        }
+
+        public void AddDeposit(CreatedDepositDTO newDeposit)
+        {
+            var deposit = new Deposit { 
+                Id = GetLastDepositID(),
```
Comment (anton-cheg) [inline App.Deposits/DepositsManager.cs]: Incorrect method

File: App.Deposits/DepositsManager.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+using System.Linq;
+
+namespace App.Deposits
+{
+    public interface IDepositsManager
+    {
+        Deposit GetDepositById(int id);
+
+        void AddDeposit(CreatedDepositDTO deposit);
+        IEnumerable<Deposit> GetAllDeposits();
```
Comment (anton-cheg) [inline App.Deposits/DepositsManager.cs]: Missing empty line between lines

File: App.Deposits/DepositsManager.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        }
+
+        public Deposit GetDepositById(int id)
+        {
+            return depositsRepository.GetDepositById(id);
+        }
+
+        public IEnumerable<Deposit> GetAllDeposits()
+        {
+            return depositsRepository.GetAllDeposit();
```
Comment (anton-cheg) [inline App.Deposits/DepositsManager.cs]: Use one style for return value from method( or => or return)

File: App.Deposits/DepositsManager.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+        private int GetLastDepositID() => depositsRepository.GetAllDeposit().OrderByDescending(x => x.Id).Select(x => x.Id).FirstOrDefault();
+
+        private int GetIDForNewDeposit() => GetLastDepositID() + 1;
+
+        public decimal AccrualСalculation(int depositId, decimal startSum, DateTime finishDate)
+        {
+            var deposit = depositsRepository.GetDepositById(depositId);
+
+            int daysAmount = (finishDate - DateTime.Now).Days;
```
Comment (anton-cheg) [inline App.Deposits/DepositsManager.cs]: use var when variables type is obvious from rigth part expressions

File: App.Deposits/DepositsManager.cs [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+            decimal sum = startSum + startSum * deposit.InterastRate * daysAmount / 365;
```
Comment (anton-cheg) [inline App.Deposits/DepositsManager.cs]: use var when variables type is obvious from rigth part expressions

File: App.Deposits/Models/CreatedDepositDTO.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace App.Deposits.Models
+{
+    public class CreatedDepositDTO
+    {
+        public string Name { get; set; }
```
Comment (anton-cheg) [inline App.Deposits/Models/CreatedDepositDTO.cs:9]: missing empty line

File: App.Deposits/Models/Deposit.cs [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace App.Deposits.Models
+{
+    public class Deposit
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public decimal InterastRate { get; set; }
```
Comment (anton-cheg) [inline App.Deposits/Models/Deposit.cs]: Missing empty lines

File: App.Deposits/Repositories/InMemmoryDepositsRepository.cs [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        Deposit GetDepositById(int id);
+
+        void AddDeposit(Deposit deposit);
+    }
+
+    public class InMemmoryDepositsRepository : IDepositsRepository, ITransientDependency
+    {
+        private static readonly List<Deposit> deposits = Init();
+
+
```
Comment (anton-cheg) [inline App.Deposits/Repositories/InMemmoryDepositsRepository.cs]: excess empty lines

File: App.Deposits/Repositories/InMemmoryDepositsRepository.cs [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        {
+            deposits.Add(deposit);
+        }
+
+        public IEnumerable<Deposit> GetAllDeposit()
+        {
+            return deposits;
+        }
+
+        public Deposit GetDepositById(int id) => deposits.Where(x => x.Id == id).FirstOrDefault();
```
Comment (anton-cheg) [inline App.Deposits/Repositories/InMemmoryDepositsRepository.cs]: Use one style for return value from method( or => or return)

**Review by anton-cheg — COMMENTED** [review]
Nice try, Dmitriy

File: App.TestModule/App.Example.csproj [hunk 10]
```diff
@@ -1,10 +1,11 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <TargetFramework>netcoreapp2.2</TargetFramework>
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.App" Version="2.2.0" />
```
Comment (DmytroOzzmos) [inline App.TestModule/App.Example.csproj]: revert

File: App.Web/App.Web.csproj [hunk 11]
```diff
@@ -20,6 +20,7 @@
     <PackageReference Include="Castle.Facilities.AspNetCore" Version="5.0.0" />
     <PackageReference Include="Castle.Windsor" Version="5.0.0" />
     <PackageReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.2.0" />
```
Comment (DmytroOzzmos) [inline App.Web/App.Web.csproj]: revert

**Review by ghost — CHANGES_REQUESTED** [review]
Please, address comments and re-request review

**Review by ghost — APPROVED** [review]

**Pull request merged**